server/src/main/java/io/druid/segment/loading/SegmentLoaderLocalCacheManager.java

```diff
@@ -270,4 +264,19 @@ public void cleanupCacheFiles(File baseFile, File cacheFile) throws IOException
       }
     }
   }
+
+  public Set<StorageLocation> getSortedSet(List<StorageLocation> locs)
```

Comment:
@drcrallen Adding same object twice would definitely be a bug but my argument was more about being defensive against these type of bugs as the locations comes from user config, what if user adds the same location twice in the list. Anyways I am OK with it being a list if it make things easier. If this becomes a `List` probably it can be de-duped initially before being used.